src/test/storage/table_test.cpp

```diff
@@ -180,4 +180,23 @@ TEST_F(StorageTableTest, MemoryUsageEstimation) {
                                                      sizeof(TransactionID) + 2 * sizeof(CommitID));
 }
 
+TEST_F(StorageTableTest, StableChunks) {
+  // Tests that pointers to a chunk remain valid even if the table grows (#1463)
+  auto table = std::make_shared<Table>(column_definitions, TableType::Data, 1);
+  table->append({100, "Hello"});
+
+  // The address of the first shared_ptr control object
+  const auto first_chunk = &table->chunks()[0];
+
+  for (auto i = 1; i < 10; ++i) {
+    table->append({i, "Hello"});
```

Comment:
I forgot to submit the PR. :-/

Reply:
> If this is the case we might consider always creating something like a vector of 20 chunks (it's just shared pointers anyway).

That's what I did above (reserving 23 chunks) - it was slower.

> I still don't feel comfortable using TBB when we are not sure what slows us down here.

We do. Accessing a concurrent vector is more expensive because you have an extra layer of indirection. Also, getting the size of it requires synchronization. The question is not why are the accesses slower, but why do we do so many accesses that it becomes significant.

I added this to the Trello list though. Once everything becomes a bit quieter after the seminar, we should look at it.